### ISSUE TITLE
Add workaround hint to upgrade guide for XML failing parser configuration 

### ISSUE THREAD

**Issue by ben-manes** [gradle/gradle#26672]
### Issue type

Missing information

### Problem description

Gradle 8.4's secure XML parsing may fail builds that include a legacy xml parser on the build environment. An old version of xerces might be transitively pulled into the classpath by a plugin such as the current Android plugin (8.3.0-alpha07), gradle-coveralls-plugin ([#115](https://github.com/kt3k/coveralls-gradle-plugin/issues/115)), previous versions of the gradle-versions-plugin ([#814](https://github.com/ben-manes/gradle-versions-plugin/issues/814)), and likely many others. The resulting `SAXNotRecognizedException` is not obvious to users or plugin authors, who might not be able to resolve this due to unmaintained transitive dependencies.

A possible workaround is to instruct the JVM to use the native provider. In OpenJDK this can be done by adding the following to `gradle.properties`:
```ini
systemProp.javax.xml.parsers.SAXParserFactory=com.sun.org.apache.xerces.internal.jaxp.SAXParserFactoryImpl
systemProp.javax.xml.transform.TransformerFactory=com.sun.org.apache.xalan.internal.xsltc.trax.TransformerFactoryImpl
systemProp.javax.xml.parsers.DocumentBuilderFactory=com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderFactoryImpl
```

Other JDKs should have similar, e.g. IBM's is supposedly
```ini
systemProp.javax.xml.parsers.SAXParserFactory=com.ibm.xml.parsers.SAXParserFactoryImpl
systemProp.javax.xml.transform.TransformerFactory=com.ibm.xml.transform.TransformerFactoryImpl
systemProp.javax.xml.parsers.DocumentBuilderFactory=com.ibm.xml.parsers.DocumentBuilderFactoryImpl
```

I cannot say this will be foolproof and not result in other errors, but it does resolve the problem in my tests and seems helpful to provide to struggling users and confused plugin authors.

### Page with the problem

https://docs.gradle.org/current/userguide/upgrading_version_8.html#xml_parsing_now_requires_recent_parsers

**Comment by ov7a** [comment]
Thank you for your interest in Gradle!

This is a valid documentation issue that we will address.

***

See also [the security advisory](https://github.com/gradle/gradle/security/advisories/GHSA-mrff-q8qj-xvg8).

**Comment by ov7a** [comment]
Fixed
https://docs.gradle.org/8.4/userguide/upgrading_version_8.html#changes_8.4

**Comment by TroutZhang** [comment]
As someone came here late and also encountered the issue when trying "profile with low over head" in AS.

The workaround really worked, but upgrading AGP to the current latest 8.7.1 also helped.

Whick led me to search and eventually found this was [fixed in AGP 8.3.0 already](https://issuetracker.google.com/u/1/issues/306301014), so the simplest way is to upgrade AGP to 8.3.0+

**Comment by TroutZhang** [comment]
@ben-manes Yeah, you're right. Maybe my wording is wrong. It should be "additional" not "alternative" hint to add.
